shardingsphere-jdbc/shardingsphere-jdbc-core/src/test/java/org/apache/shardingsphere/driver/jdbc/adapter/ConnectionAdapterTest.java

```diff
@@ -172,6 +169,17 @@ private void assertClose(final ShardingSphereConnection actual) {
         assertTrue(cachedConnections.isEmpty());
     }
     
+    @Test
+    public void assertCloseShouldNotClearTransactionType() throws SQLException {
+        TransactionTypeHolder.set(TransactionType.XA);
+        TransactionType currentTransactionType = TransactionTypeHolder.get();
+        try (ShardingSphereConnection actual = getShardingSphereDataSource().getConnection()) {
+            actual.createStatement().executeQuery(sql);
+            actual.close();
```

Comment:
actual.close() can be removed, try() statement will invoke close method automatically